vllm/distributed/parallel_state.py

```diff
@@ -272,17 +272,17 @@ def all_reduce(self, input_: torch.Tensor) -> torch.Tensor:
 
         # Bypass the function if we are using only 1 GPU.
         if self.world_size == 1:
-            return input_
+            return input_.clone()
```

Comment:
this clone can hurt performance. why do we need this?

Reply:
This was part of the custom ar PR that didn't get removed.  I'll fix it.